### PR TITLE
feat(storage.cold-archive): add container list view & some refactoring.

### PR DESCRIPTION
**Pull request by brofik**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/MANAGER-9069`  
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #feat/MANAGER-10190 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    data-track-on="click"
+                    data-track-name="PCI_PROJECTS_CONTAINERS_ADD"
+                    data-track-type="navigation"
+                >
+                    <span
+                        class="oui-icon oui-icon-add pr-1"
+                        aria-hidden="true"
+                    ></span>
+                    <span
+                        data-ng-bind="'pci_projects_project_storages_cold_archive_containers_add_container_label' | translate"
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html:88]: Suggestion: Use: data-translate

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.component.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export default {
+  controller,
+  template,
+  bindings: {
+    containers: '<',
+    refreshContainers: '<',
+    goToAddColdArchive: '<',
+    goToAddUserContainer: '<',
+    archiveContainer: '<',
+    restaureContainer: '<',
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.component.js:13]: Spell mistake: restaureContainer -> restoreContainer ?

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html (hunk 1 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html]: Done.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.component.js (hunk 2 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.component.js]: yes.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.controller.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    const { componentClass } = CONTAINER_STATUS_OPTIONS[status];
+    return componentClass;
+  }
+
+  getAction(action) {
+    const ACTIONS = {
+      add_user: this.goToAddUserContainer,
+      archive: this.archiveContainer,
+      restore: this.restoreContainer,
+      delete: this.deleteContainer,
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.controller.js:51]: Actions here are not matching with actions in CONTAINER_STATUS_OPTIONS constant. Can you please check ?

Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.controller.js:51]: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.constants.js

**Review by sachinrameshn — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.controller.js (hunk 3 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.controller.js:50]: This issue was fixed in an incoming PR. You're right I need to fix it in this PR.
Done.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/users/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -13,6 +14,6 @@
   "pci_projects_project_storages_containers_users_add_user": "Ajouter un utilisateur",
   "pci_projects_project_storages_containers_users_import_success_message": "Le téléchargment du fichier JSON lié à l’utilisateur {{user}} a été effectué avec succès.",
   "pci_projects_project_storages_containers_users_import_error_message": "Une erreur est survenue lors du téléchargement du fichier JSON lié à l’utilisateur : {{ message }}.",
-  "pci_projects_project_storages_containers_users_show_secret_key_success": "La clé secrète de votre utilisateur {{user}} est {{secret}}.",
+  "pci_projects_project_storages_containers_users_show_secret_key_success": "La clé secrète de votre utilisateur <strong>{{user}}</strong> est {{secret}}.",
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/users/translations/Messages_fr_FR.json:17]: **nitpick:** `<strong>` html tag can be avoided in translation files by including it in the `user` in HTML with `data-translate-value`

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -1,7 +1,30 @@
...
+
+  "pci_projects_project_storages_cold_archive_container_label": "Containers",
+  "pci_projects_project_storages_cold_archive_containers_creation_date_label": "Date de création",
+  "pci_projects_project_storages_cold_archive_containers_add_container_label": "Créer une archive",
+  "pci_projects_project_storages_cold_archive_containers_status_label": "Status",
+  "pci_projects_project_storages_containers_status_none": "Actif",
+  "pci_projects_project_storages_containers_status_restored": "Restauré",
+  "pci_projects_project_storages_containers_status_restoring": "En cours de restauration",
+  "pci_projects_project_storages_containers_status_archived": "Archivé",
+  "pci_projects_project_storages_containers_status_archiving": "En cours d'achivage",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/translations/Messages_fr_FR.json:24]:
```suggestion
  "pci_projects_project_storages_containers_status_archiving": "En cours d'archivage",
```

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html [hunk 6]
```diff
@@ -0,0 +1,104 @@
+<div data-ui-view>
+    <div>
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html:2]: probably useless `div` element ?

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js [hunk 7]
```diff
@@ -219,4 +235,17 @@ export default class PciStoragesColdArchiveService {
...
       .then(({ data }) => data);
   }
+
+  /**
+   * Mock API Call
+   * [TODO]: To be deleted when about to merge with the release branch.
+   */
+  getAllColdArchives() {
+    return this.$http
+      .get('http://localhost:3200/mock_api/storage/cold/mock/all')
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js:245]: To remove and use API instead.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js [hunk 8]
```diff
@@ -219,4 +235,17 @@ export default class PciStoragesColdArchiveService {
...
+      .then(({ data }) => {
+        // eslint-disable-next-line no-shadow
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js:247]: probably we could avoid this.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/users/translations/Messages_fr_FR.json (hunk 4 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/users/translations/Messages_fr_FR.json:17]: Done

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/translations/Messages_fr_FR.json (hunk 5 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/translations/Messages_fr_FR.json:24]: Done

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html (hunk 6 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/containers/containers.html:2]: Done

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js (hunk 8 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js:247]: Done

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js (hunk 7 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/cold-archive.service.js:245]: Useless function.
Removed.

**Review by mohammed-zahaf — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Pull request merged**